sitegen/src/main/java/io/helidon/sitegen/asciidoctor/Block.java

```diff
@@ -0,0 +1,351 @@
+ * // _include::6-9:Include2.adoc
+ * ----
+ * // This is un-included source
+ * src inc 1.1
+ * src inc 1.2
+ * src inc 1.3
+ * src inc 1.4
+ * More un-included source
+ * src inc 2.1
+ * src inc 2.2
+ * src inc 2.3
+ * src inc 2.4
+ * Further content
+ * ----
+ * }
+ * </pre>
+ * <h2>Creating {@code Block} Instances by Parsing AsciiDoc</h2>
+ * The {@link #consumeBlock} method reads content and builds an instance
+ * representing the corresponding block, advancing the current line indicator
+ * for the content as it does so.
+ * <p>
+ * Note that this method handles the original AsciiDoc form (with
+ * {@code include::} directives in the body), pure numbered format, and a hybrid
+ * in which the user has added AsciiDoc {@code include::} directives in the body
+ * of the numbered form.
+ */
+public class Block {
+
+    private static final Set<String> BLOCK_DELIMITERS
+            = new HashSet<>(Arrays.asList(
+                    new String[]{"====", // example
+                        "----", // listing, source
+                        "...." // listing
+                }));
+
+    private static final Set<String> BLOCK_INTRODUCERS
+            = new HashSet<>(Arrays.asList(
+                    new String[]{"source", "listing", "example"}));
+
+    /**
+     * Creates a Block by consuming the input text, advancing {@code lineNumber}
+     * so it points just past the end of the block's ending delimiter.
+     *
+     * @param content lines containing AsciiDoc
+     * @param lineNumber line number at which to begin processing the block
+     * block
+     * @return a new Block describing the block
+     */
+    static Block consumeBlock(List<String> content, AtomicInteger lineNumber) {
+        Block sb = new Block();
+        sb.prepare(content, lineNumber);
+        return sb;
+    }
+
+    /**
+     * Returns whether the specified line represents the start of a block.
+     *
+     * @param line the line to check
+     * @return true if the line starts a block; false otherwise
+     */
+    static boolean isBlockStart(String line) {
+        return BLOCK_INTRODUCERS.stream()
+                .anyMatch(intro -> line.startsWith("[" + intro));
+    }
+
+    private static boolean isBlockDelimiter(String line) {
+        return BLOCK_DELIMITERS.stream()
+                .anyMatch(delimiter -> line.equals(delimiter));
+    }
+
+    private String blockDecl = null;
+    private String delimiter = null;
+    private final List<Include> includes = new ArrayList<>();
+    private final List<String> body = new ArrayList<>();
+
+    private List<String> preamble = new ArrayList<>();
+
+    /**
+     * Formats the block using numbered include comments in the preamble and the
+     * actual inserted text in the body.
+     *
+     * @return block with (if needed) numbered include comments in the preamble
+     */
+    List<String> asBlockWithNumberedIncludes() {
+        return asBlock(() -> {
+            List<String> result = new ArrayList<>();
```

Comment:
Could this be instead something like this?
```java
return asBlock (() -> includes.stream()
    .map(Include::asNumberedAsciidocInclude)
    .collect(Collectors.toList()));
```

Reply:
Yes!